File: SelectSort.java
public class SelectSort{
	public static void main(String[] args){
		int [] array = { 2,5,6,23,1,10};
		int i, j, min, tmp;
		for(i = 0 ; i < array.length; i++){
			min = i;
			for(j = i+1 ; j < array.length ; j++){ //i+1 이므로 length -1 하지 않는다.
				if(array[j] < array[min])
					min = j;
			}
			tmp = array[i];
			array[i] = array[min];
			array[min] = tmp;
		}

		for(int h = 0 ; h < array.length; h++){
			System.out.println(array[h]+ "");
		}

	}
}

File: SelectSort.js
// SelectSort.js

Array.prototype.SelectSort = function(){
	var i, j, min;
	var size = this.length;
	for(i=0; i < size; i++){
		min = i;
		for( j=i+1; j<size; j++){
			if(this[j] < this[min]){
				min = j;
			}
		}
		var tmp = this[i]
		this[i] = this[min]
		this[min] = tmp;
	}
	return this
}

var array = [6,5,4,3,2,1]

console.log(array.SelectSort())